pydantic/_internal/_generate_schema.py

```diff
@@ -1837,16 +1837,17 @@ def _dataclass_schema(
                 self.defs.definitions[dataclass_ref] = schema
                 return core_schema.definition_reference_schema(dataclass_ref)
 
-    def _callable_schema(self, function: Callable[..., Any]) -> core_schema.CallSchema:
+            # Type checkers seem to assume ExitStack may suppress exceptions and therefore
+            # control flow can exit the `with` block without returning.
+            assert False, 'Unreachable'
+
+    def _call_schema(self, function: ValidateCallSupportedTypes) -> core_schema.CallSchema:
         """Generate schema for a Callable.
 
         TODO support functional validators once we support them in Config
         """
         sig = signature(function)
 
-        globalns, localns = self._types_namespace
-        type_hints = _typing_extra.get_function_type_hints(function, globalns=globalns, localns=localns)
```

Comment:
I'm concerned about the removal of this, cc @Viicos.

Reply:
This is fine because the `self._resolve_forward_ref` will take care of it (and we used `ns_for_function` earlier on to get the ns)